src/engine/sidechain/engineshoutcast.cpp

```diff
@@ -62,47 +62,45 @@ EngineShoutcast::EngineShoutcast(ConfigObject<ConfigValue>* _config)
           m_protocol_is_icecast1(false),
           m_protocol_is_icecast2(false),
           m_protocol_is_shoutcast(false),
-          m_ogg_dynamic_update(false) {
-
-#ifndef __WINDOWS__
-    // Ignore SIGPIPE signals that we get when the remote streaming server
-    // disconnects.
-    signal(SIGPIPE, SIG_IGN);
-#endif
-
-    m_pShoutcastStatus->set(SHOUTCAST_DISCONNECTED);
-    m_pShoutcastNeedUpdateFromPrefs = new ControlObject(
-            ConfigKey(SHOUTCAST_PREF_KEY,"update_from_prefs"));
-    m_pUpdateShoutcastFromPrefs = new ControlObjectSlave(
-            m_pShoutcastNeedUpdateFromPrefs->getKey());
+          m_ogg_dynamic_update(false),
+          m_threadWaiting(false),
+          m_pOutputFifo(NULL) {
+    const bool persist = true;
+    m_pShoutcastEnabled = new ControlPushButton(
+            ConfigKey(SHOUTCAST_PREF_KEY,"enabled"), persist);
+    m_pShoutcastEnabled->setButtonMode(ControlPushButton::TOGGLE);
+    connect(m_pShoutcastEnabled, SIGNAL(valueChanged(double)),
+            this, SLOT(slotEnableCO(double)));
+
+    m_pStatusCO = new ControlObject(ConfigKey(SHOUTCAST_PREF_KEY, "status"));
+    m_pStatusCO->connectValueChangeRequest(
+            this, SLOT(slotStatusCO(double)));
+    m_pStatusCO->setAndConfirm(STATUSCO_UNCONNECTED);
+
+    setState(NETWORKSTREAMWORKER_STATE_INIT);
 
     // Initialize libshout
     shout_init();
 
     if (!(m_pShout = shout_new())) {
         errorDialog(tr("Mixxx encountered a problem"), tr("Could not allocate shout_t"));
-        return;
     }
 
     if (!(m_pShoutMetaData = shout_metadata_new())) {
         errorDialog(tr("Mixxx encountered a problem"), tr("Could not allocate shout_metadata_t"));
-        return;
     }
+
     if (shout_set_nonblocking(m_pShout, 1) != SHOUTERR_SUCCESS) {
         errorDialog(tr("Error setting non-blocking mode:"), shout_get_error(m_pShout));
-        return;
     }
 }
 
 EngineShoutcast::~EngineShoutcast() {
-    if (m_encoder) {
-        m_encoder->flush();
-        delete m_encoder;
-    }
+    m_pShoutcastEnabled->set(0);
+    m_readSema.release();
+    wait(); // until the thread ends.
```

Comment:
A timout here doe not make sense. Since a zombie thread makes Mixxx hang at exit as well. 
Exit works reliably here on my side, even if Mixxx is still broadcasting.
Are you able to reproduce the issue?   
If the issue persists, we should placed a timout on the blocking call inside the thread. 
Any idea where?